test/test_eos.py

```diff
@@ -24,47 +24,367 @@
+
+    discr = EagerDGDiscretization(actx, mesh, order=order)
+    ones = discr.zeros(actx) + 1.0
+
+    # Pyrometheus initialization
+    mech_cti = get_mechanism_cti(mechname)
+    cantera_soln = cantera.Solution(phase_id="gas", source=mech_cti)
+    pyro_obj = pyro.get_thermochem_class(cantera_soln)(actx.np)
+
+    nspecies = pyro_obj.num_species
+    print(f"PrometheusMixture::NumSpecies = {nspecies}")
+
+    tempin = 1500.0
+    pressin = cantera.one_atm
+    print(f"Testing (t,P) = ({tempin}, {pressin})")
+
+    # Homogeneous reactor to get test data
+    equiv_ratio = 1.0
+    ox_di_ratio = 0.21
+    stoich_ratio = 0.5
+    i_fu = cantera_soln.species_index("H2")
+    i_ox = cantera_soln.species_index("O2")
+    i_di = cantera_soln.species_index("N2")
+    x = np.zeros(shape=(nspecies,))
+    x[i_fu] = (ox_di_ratio*equiv_ratio)/(stoich_ratio+ox_di_ratio*equiv_ratio)
+    x[i_ox] = stoich_ratio*x[i_fu]/equiv_ratio
+    x[i_di] = (1.0-ox_di_ratio)*x[i_ox]/ox_di_ratio
+
+    cantera_soln.TPX = tempin, pressin, x
+    #    cantera_soln.equilibrate("UV")
+    can_t, can_rho, can_y = cantera_soln.TDY
+    #    can_p = cantera_soln.P
+
+    reactor = cantera.IdealGasConstPressureReactor(cantera_soln)
+    sim = cantera.ReactorNet([reactor])
+    time = 0.0
+    for step in range(50):
+        time += 1.0e-6
+        sim.advance(time)
+
+        # Cantera kinetics
+        can_r = reactor.kinetics.net_rates_of_progress
+        can_omega = reactor.kinetics.net_production_rates
+
+        # Get state from Cantera
+        can_t = reactor.T
+        can_rho = reactor.density
+        can_y = reactor.Y
+        print(f"can_y = {can_y}")
+
+        tin = can_t * ones
+        rhoin = can_rho * ones
+        yin = can_y * ones
+
+        # Prometheus kinetics
+        pyro_c = pyro_obj.get_concentrations(rhoin, yin)
+        print(f"pyro_conc = {pyro_c}")
+
+        pyro_r = pyro_obj.get_net_rates_of_progress(tin, pyro_c)
+        pyro_omega = pyro_obj.get_net_production_rates(rhoin, tin, yin)
+
+        # Print
+        print(f"can_r = {can_r}")
+        print(f"pyro_r = {pyro_r}")
+        abs_diff = discr.norm(pyro_r - can_r, np.inf)
+        if abs_diff > 1e-14:
+            for i, rate in enumerate(can_r):
+                min_r = (np.abs(can_r)).min()
+                if min_r > 0:
+                    assert discr.norm((pyro_r - can_r) / can_r, np.inf) < rate_tol
+                else:
+                    assert discr.norm(pyro_r, np.inf) < rate_tol
+
+        print(f"can_omega = {can_omega}")
+        print(f"pyro_omega = {pyro_omega}")
+        for i, omega in enumerate(can_omega):
+            omin = np.abs(omega).min()
+            if omin > 1e-12:
+                assert discr.norm((pyro_omega[i] - omega) / omega, np.inf) < 1e-8
+            else:
+                assert discr.norm(pyro_omega[i], np.inf) < 1e-12
+
+
+@pytest.mark.parametrize("dim", [1, 2, 3])
+def test_idealsingle_lump(ctx_factory, dim):
```

Comment:
Would it make sense to add a test with a flow problem with chemistry? Do we have any where the solution is known? If we have to do self-convergence, would that be too expensive?

Reply:
Duh. #237 and #238. Nvm.

Reply:
Hmm, those are examples though, and they don't really check for a specific outcome. Is there anything where we could check that something specific happens?

Reply:
A super easy test to add would be one where there is a uniform mixture and uniform flow, with and without chemistry.  The RHS should be 0's in both cases. This sounds trivial, but it'd be hard to get right by fluke (unless you just return 0's always - which we know we don't).

For most of the examples that I add, the thing will actually check against the exact solution and fail if it doesn't ~match.  You are right that I didn't do that for the chem example and (effectively) not for the mixture example.  For the chem case, I am not sure I can come up with something reasonable on the short-term.   For the mixture case, I should be able to figure something out that is non-trivial.

Lemme think on this a bit.

Reply:
@MTCam Had any luck with coming up with a test? (FWIW, I'm OK if the answer is no... just asking since this conversation is kind of just dangling unresolved.)

Reply:
Actually have had quite a few discussions about this with @ecisneros8.  Although Esteban has some ideas, neither of us have anything that doesn't carry some hefty expenses; computational, development, and data.   

But we aren't giving up.  Made an issue on this: (#309)